### PR TITLE
feat: [IOPID-1569] Add new DS to EmailAlreadyTakenScreen

### PR DESCRIPTION
**Pull request by Ladirico**
## Short description
Add new ds to EmailAlreadyTakenScreen using existent component OperationResultScreenContent. 

## List of changes proposed in this pull request
- Integration of new DS using OperationResultScreenContent
- Addition of the prop `isHeaderVisible ` on OperationResultScreenContent. If header is visible add bottom margin to vertically centre the component in the screen. 
- Update jest snapshots

### A11Y
<p>

| 🍏 A11Y iOS 🍏 | 🤖 A11Y Android 🤖 |
| - | - | 
| <video src="https://github.com/pagopa/io-app/assets/83651704/70b190db-be97-4a43-ad58-b8b4a8575640" /> | <video src="https://github.com/pagopa/io-app/assets/83651704/d2b17ce8-f01e-4bc2-ac0d-238e3b9bfb90" /> |

</p>

## How to test
Run the application and in io-dev-api-server add is_email_already_taken: true

### CODE REVIEW

**Comment by Ladirico** [comment]
@mariateresaventura need copy review! 😁

File: ts/components/screens/OperationResultScreenContent.tsx [hunk 1]
```diff
@@ -83,6 +85,7 @@ const OperationResultScreenContent = ({
         /* Android fallback because `centerContent` is only an iOS property */
         Platform.OS === "android" && styles.wrapper_android
       ]}
+      style={{ marginTop: headerHeight ? -headerHeight : undefined }}
```
Comment (shadowsheep1) [inline ts/components/screens/OperationResultScreenContent.tsx:88]: Do we prefer to act on `margins` or on `SafeAreaView` edges (e.g. `edges={["bottom"]}`) when we use this component in a screen showing the react-navigation header? What do you think?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: ts/components/screens/OperationResultScreenContent.tsx (hunk 1 above)
Comment (Ladirico) [reply ts/components/screens/OperationResultScreenContent.tsx]: yess! Done!

**Comment by mariateresaventura** [comment]
> @mariateresaventura need copy review! 😁

done!

File: ts/components/screens/OperationResultScreenContent.tsx [hunk 2]
```diff
@@ -73,9 +75,14 @@ const OperationResultScreenContent = ({
   action,
   secondaryAction,
   children,
-  testID
+  testID,
+  isVisibleHeader
```
Comment (shadowsheep1) [inline ts/components/screens/OperationResultScreenContent.tsx:79]:
```suggestion
  isHeaderVisible
```

`isVisibleHeader`, while understandable, inverts the typical order of descriptors and subjects. The preferred choice would be `isHeaderVisible`. This naming convention follows a common pattern in programming where the "is" prefix indicates a boolean value, followed by the subject (`Header`) and the property being described (`Visible`). What do you think?

File: ts/components/screens/OperationResultScreenContent.tsx [hunk 3]
```diff
@@ -73,9 +75,14 @@ const OperationResultScreenContent = ({
...
 }: React.PropsWithChildren<OperationResultScreenContentProps>) => (
-  <SafeAreaView style={styles.container} testID={testID}>
+  <SafeAreaView
+    edges={isVisibleHeader ? ["bottom"] : undefined}
```
Comment (shadowsheep1) [inline ts/components/screens/OperationResultScreenContent.tsx:82]:
```suggestion
    edges={isHeaderVisible ? ["bottom"] : undefined}
```

File: ts/screens/profile/mailCheck/EmailAlreadyTakenScreen.tsx [hunk 4]
```diff
@@ -65,55 +57,58 @@ const EmailAlreadyTakenScreen = () => {
...
-            </Body>
-          </Body>
-        </View>
-        <FooterWithButtons
-          type={"SingleButton"}
-          leftButton={continueButtonProps}
-        />
-      </SafeAreaView>
-    </BaseScreenComponent>
+      isVisibleHeader={true}
```
Comment (shadowsheep1) [inline ts/screens/profile/mailCheck/EmailAlreadyTakenScreen.tsx:110]:
```suggestion
      isHeaderVisible ={true}
```

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: ts/components/screens/OperationResultScreenContent.tsx (hunk 2 above)
Comment (Ladirico) [reply ts/components/screens/OperationResultScreenContent.tsx:79]: Yes! Sure! Thanks for the suggestion!

Comment (Ladirico) [reply ts/components/screens/OperationResultScreenContent.tsx]: done!

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**